src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterEnum.cs

```diff
@@ -39,17 +49,54 @@ public override bool TryRead(Type valueType, ref Utf8JsonReader reader, out TVal
                 return false;
             }
 
-            if (s_isUint64)
+            if (s_isUInt64 && reader.TryGetUInt64(out ulong uint64))
             {
-                if (reader.TryGetUInt64(out ulong ulongValue))
-                {
-                    value = (TValue)Enum.ToObject(valueType, ulongValue);
-                    return true;
-                }
+                value = (TValue)Enum.ToObject(valueType, uint64);
+                return true;
+            }
+
+            if (s_isInt64 && reader.TryGetInt64(out long int64))
+            {
+                value = (TValue)Enum.ToObject(valueType, int64);
+                return true;
+            }
+
+            if (s_isUInt32 && reader.TryGetUInt32(out uint uint32))
+            {
+                value = (TValue)Enum.ToObject(valueType, uint32);
+                return true;
+            }
+
+            if (s_isInt32 && reader.TryGetInt32(out int int32))
+            {
+                value = (TValue)Enum.ToObject(valueType, int32);
+                return true;
+            }
+
+            // When utf8reader/writer will support all primitive types we should remove custom bound checks
+            // https://github.com/dotnet/corefx/issues/36125
+
+            if (s_isUInt16 && reader.TryGetUInt32(out uint uint16) && uint16 >= ushort.MinValue && uint16 <= ushort.MaxValue)
+            {
+                value = (TValue)Enum.ToObject(valueType, uint16);
+                return true;
+            }
+
+            if (s_isInt16 && reader.TryGetInt32(out int int16) && int16 >= short.MinValue && int16 <= short.MaxValue)
```

Comment:
A `IsRange` helper can help reduce the number of comparisons.
https://github.com/dotnet/corefx/blob/6d37591d13dd58abc90b7f15b707f605e809be29/src/System.Text.Json/src/System/Text/Json/JsonHelpers.cs#L47

Reply:
oh thanks missed that.

Reply:
done